nominal/core/unit.py

```diff
@@ -0,0 +1,66 @@
+import dataclasses
+
+from typing_extensions import Self
+
+from .._api.combined import datasource_api, scout_asset_api, scout_run_api, scout_units_api
+
+
+@dataclasses.dataclass(frozen=True)
+class Unit:
+    """Plain python wrapper to encapsulate the data from scout_units_api.Unit.
+
+    This is primarily used when setting or retrieving the units of a channel within a dataset.
+    """
+
+    # Plain english name of the unit (e.g. 'coulomb')
+    name: str
```

Comment:
you should actually turn these into docstrings on the attribute, they'll show up in the docs!